patches/tModLoader/Terraria/ModLoader/AccessorySlotLoader.cs

```diff
@@ -0,0 +1,426 @@
+			bool flag = flag3 && !Main.mouseItem.IsAir;
+			int xLoc1 = xLoc - 47 * (Math.Abs(context) - 10);
+
+			if (!skipCheck && Main.mouseX >= xLoc1 && (float)Main.mouseX <= (float)xLoc1 + (float)TextureAssets.InventoryBack.Width() * Main.inventoryScale && Main.mouseY >= yLoc
+				&& (float)Main.mouseY <= (float)yLoc + (float)TextureAssets.InventoryBack.Height() * Main.inventoryScale && !PlayerInput.IgnoreMouseInterface) {
+
+				Player.mouseInterface = true;
+				Main.armorHide = true;
+				ItemSlot.OverrideHover(items, Math.Abs(context), slot);
+
+				if (!flag) {
+					if (Math.Abs(context) == 12) {
+						if (Main.mouseRightRelease && Main.mouseRight)
+							ItemSlot.RightClick(items, Math.Abs(context), slot);
+
+						ItemSlot.LeftClick(items, context, slot);
+					}
+					else if (Math.Abs(context) == 11) {
+						ItemSlot.LeftClick(items, context, slot);
+						ItemSlot.RightClick(items, Math.Abs(context), slot);
+					}
+					else if (Math.Abs(context) == 10) {
+						ItemSlot.LeftClick(items, context, slot);
+					}
+				}
+
+				ItemSlot.MouseHover(items, Math.Abs(context), slot);
+			}
+			DrawRedirect(items, context, slot, new Vector2(xLoc1, yLoc));
+		}
+
+		internal void DrawRedirect(Item[] inv, int context, int slot, Vector2 location) {
+			if (context < 0) {
+				Get(slot).DrawModded(inv, context, slot, location);
+			} else {
+				ItemSlot.Draw(Main.spriteBatch, inv, context, slot, location);
+			}
+		}
+
+		// Functionality Related Code /////////////////////////////////////////////////////////////////////
+
+		public bool ModdedIsAValidEquipmentSlotForIteration(int index) => Get(index).IsEnabled();
+
+		public bool ModdedCanSlotBeShown(int index) => Get(index).IsVisibleWhenNotEnabled();
+
+		public bool ModdedSkipUIDraw(int index) => Get(index).SkipUIDrawWileTrue;
+
+		public bool CanAcceptItem(int index, Item checkItem) => Get(index).CanAcceptItem(checkItem);
+
+		public void CustomUpdateEquips(int index) => Get(index).ApplyEquipEffects();
+
+		/// <summary>
+		/// Checks if the provided item can go in to the provided slot. 
+		/// Includes checking if the item already exists in either of Player.Armor or ModSlotPlayer.exAccessorySlot
+		/// Invokes directly ItemSlot.AccCheck & ModSlot.CanAcceptItem
+		/// </summary>
+		public bool ModSlotCheck(Item checkItem, int slot) => CanAcceptItem(slot, checkItem) &&
+			!ItemSlot.AccCheck(Player.armor.Concat(ModSlotPlayer(Player).exAccessorySlot).ToArray(), checkItem, slot + Player.armor.Length);
+
+		/// <summary>
+		/// After checking for empty slots in ItemSlot.AccessorySwap, this allows for changing what the target slot will be if the accessory isn't already equipped.
+		/// DOES NOT affect vanilla behaviour of swapping items like for like where existing in a slot
+		/// </summary>
+		public void ModifyDefaultSwapSlot(Item item, ref int accSlotToSwapTo) {
+			for (int num = 0; num < ModSlotPlayer(Player).SlotCount(); num++) {
```

Comment:
It might be nicer to run this iteration in reverse order? Otherwise, if I have 2 slots which are both 'priority' for an item, it will pick the last one. (Unless I do some weird check on `accSlotToSwapTo`)